### PR TITLE
add Customer : model, validate mobile and home numer and email Address.

### PR DESCRIPTION
**Pull request by PES34**
Add tests

### CODE REVIEW

File: src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.stereotype.Service;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Service
+public class CustomerService {
+    private final CustomerRepository customerRepository;
+
+    public CustomerService(CustomerRepository customerRepository) {
```
Comment (mjasinskee) [inline src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java:13]: Please correct element ordering. 
- fields declaration
- constructors
- public methods
- private methods

File: src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        };
+
+        if (!validateMobileNumber(customer.getMobileNumber())) {
+            throw new MobileNumberNotValidException("Mobile number not valid");
+        };
+
+        return customerRepository.save(customer);
+    }
+
+    public boolean validateEmailAddress(String emailAddress) {
```
Comment (mjasinskee) [inline src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java:37]: does it have to be public?

File: src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+        boolean valid = EmailValidator.getInstance().isValid(emailAddress);
+        if (valid) {
+            return true;
+        }
+        return false;
+    }
+
+    public boolean validateMobileNumber(String mobileNumber) {
```
Comment (mjasinskee) [inline src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java:46]: does it have to be public?

File: src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        regexPatternMobile = Pattern.compile("[0-9]{3}-[0-9]{3}-[0-9]{3}$");
+        regMatcher = regexPatternMobile.matcher(mobileNumber);
+        if (regMatcher.matches()) {
+            return true;
+        }
+        return false;
+    }
+
+    public boolean validateHomeNumber(String mobileNumber) {
```
Comment (mjasinskee) [inline src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java:55]: does it have to be public?

File: src/test/java/com/inqoo/trainingservice/app/service/CustomerServiceIT.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.boot.test.context.SpringBootTest;
+
+import javax.transaction.Transactional;
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+@Transactional
+public class CustomerServiceIT {
```
Comment (mjasinskee) [inline src/test/java/com/inqoo/trainingservice/app/service/CustomerServiceIT.java:19]: Please add test cases for happy paths also

File: src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+        this.customerRepository = customerRepository;
+    }
+
+    private Pattern regexPatternMobile;
+    private Pattern regexPatternHomeNumber;
+    private Matcher regMatcher;
+
+    public Customer saveNewCustomer(Customer customer) {
+        if (!validateEmailAddress(customer.getEmailAddress())) {
+            throw new EmailNotValidException("Mail not valid");
```
Comment (mjasinskee) [inline src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java:23]: Those messages can be moved into exception directly

**Review by mjasinskee — APPROVED** [review]
please fix issues before merge

File: src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java (hunk 1 above)
Comment (mb83pl) [reply src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java:13]: done

File: src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java (hunk 2 above)
Comment (mb83pl) [reply src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java:37]: done

File: src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java (hunk 3 above)
Comment (mb83pl) [reply src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java:46]: done

File: src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java (hunk 4 above)
Comment (mb83pl) [reply src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java:55]: done

File: src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java (hunk 6 above)
Comment (mb83pl) [reply src/main/java/com/inqoo/trainingservice/app/customer/CustomerService.java:23]: done

File: src/test/java/com/inqoo/trainingservice/app/service/CustomerServiceIT.java (hunk 5 above)
Comment (mb83pl) [reply src/test/java/com/inqoo/trainingservice/app/service/CustomerServiceIT.java:19]: done

**Pull request merged**